Makefile

```diff
@@ -1,16 +1,73 @@
+.DEFAULT: help ;
+
+help:
+	# The following make commands are available:
+	# clean             - remove all generated files and directories
+	# all               - install locally, prepare for PyPI, run tests, run speed test
+	# deps              - install required developer python dependencies
+	# annotation        - rebuild annotation.html with colored cython code generator report
+	# build_pypi        - PyPI archive preparation
+	# upload_pypi       - upload built module to PyPI
+	# install_system    - build the module and install to the system
+	# uninstall_system  - uninstall streaming_form_data from the system
+	# install_local     - build the module in the current directory
+	#                     it will be available for import from the project root directory
+	# test              - run tests and check code formatting
+	# profile           - gather library function call statistics (time, count, ...)
+	# speedtest         - calculate library bandwidth
+
 clean:
-	rm -f dist/*.tar.gz
-	rm -f dist/*.whl
+	rm -f streaming_form_data/*.pyd
 	rm -rf build
+	rm -rf dist
+	rm -rf streaming_form_data.egg-info
 
-build:
-	python setup.py sdist
+build_pypi_output = dist/build_pypi.touch
+install_local_output = streaming_form_data.egg-info/PKG-INFO
+
+deps: requirements.dev.txt
+	python -m pip install -r requirements.dev.txt
+
+all: deps annotation build_pypi install_local test speedtest ;
 
-upload: build
+annotation: annotation.html ;
+
+build_pypi: annotation $(build_pypi_output) ;
+
+upload_pypi: build_pypi
 	twine upload dist/*
 
-test:
-	PYTHONPATH=. py.test
+install_system:
+	python -m pip install .
+
+uninstall_system:
+	python -m pip uninstall streaming_form_data
+
+install_local: annotation $(install_local_output) ;
+
+test: install_local
```

Comment:
The `install_local` dependency here means that every time the tests are run, the annotations file would also be updated. Is this intended?